### ISSUE TITLE
Monitor sometimes hangs

### ISSUE THREAD

**Issue by maboleth** [stsdc/monitor#140]
Elementary os latest ver. When I try to open it or to see the resources, it hangs and does not open. It happens every now and then (usually when needed!).

It seems buggy when it should be rock solid. I thought it was meant to replace processes of task manager in Windows, where you can easily kill process, effectively and ultra stable.

But alas it's not. Uninstall.

**Comment by stsdc** [comment]
Please elaborate: when it hangs? Did You look at #88 issue?

There is no major version of this app, so this indicates that it may be buggy. No one pays me nor to anyone of contributors of the Monitor, we do this in our spare time.

You can help us to make it more stable by contributing.

**Comment by maboleth** [comment]
Alright I will install it again.

As a veteran PC/Win user, I tried linux numerous times and I swear every time I tried I got a major slap in the face. Bugs, bugs and bugs. Granted, there's less now than it used to be in mid 2000's and most of the stuff works ootb. BUT when they dont, it's such annoyance. Like I cannot delete email because the Elementary Mail app refuses to do so. File a bug report. I cannot use monitor because it hangs. File a bug report. I cannot properly install Wine32 because of dependencies that cannot be met without seriously compromising the OS. Bug report already there. One player took 50min to synchronize my music files while Foobar2000 in Windows takes 2sec. File a bug report. 
OMG.

Sorry for the rant, but yes, I'm getting annoyed again.

It hangs when one app is not responding so I want to kill the process. Usually I cannot because, well, monitor hangs then as well. Next time I will start in terminal to see what message it gives to pinpoint the issue.

**Comment by maboleth** [comment]
It hangs, when I want to start it in Terminal I get: Failed to register: Timeout was reached

After I kill it with killall and start again I get:

(com.github.stsdc.monitor:13146): Gtk-WARNING **: 22:23:20.267: Refusing to add the same cell renderer to a GtkCellAreaBox twice
** (com.github.stsdc.monitor:13146): CRITICAL **: 22:23:20.361: monitor_model_helper_set_static_columns: assertion 'name != NULL' failed

It's basically dead now, cannot move the window or anything.

**Comment by stsdc** [comment]
Well that's unusual.
Please run it with `G_MESSAGES_DEBUG=all ./com.github.stsdc.monitor` so You will see more info.

**Comment by stsdc** [comment]
If You need more stability right now, use Gnome System Monitor.

**Comment by maboleth** [comment]
I'm sorry but when I type:

com.github.stsdc.monitor G_MESSAGES_DEBUG=all ./com.github.stsdc.monitor

I get this:
(com.github.stsdc.monitor:15331): GLib-GIO-CRITICAL **: 22:40:49.838: This application can not open files.

Is there something I'm doing wrong here?

Thanks for gnome sys mon. Will take a look at that too!

**Comment by stsdc** [comment]
You prepended command with `com.github.stsdc.monitor`. You should run instead: `G_MESSAGES_DEBUG=all ./com.github.stsdc.monitor`

**Comment by maboleth** [comment]
Ah, sorry. I had to remove that "./ " in order to start.

Running this and having Monitor hung I got trillions of these messages:
** (com.github.stsdc.monitor:16628): DEBUG: 23:01:27.956: GenericModel.vala:226: add_process_to_row 0
** (com.github.stsdc.monitor:16628): DEBUG: 23:01:27.956: GenericModel.vala:226: add_process_to_row 0
** (com.github.stsdc.monitor:16628): DEBUG: 23:01:27.956: GenericModel.vala:226: add_process_to_row 0
** (com.github.stsdc.monitor:16628): DEBUG: 23:01:27.956: GenericModel.vala:226: add_process_to_row 0

And in the mean time the app was unresponsive.
